### PR TITLE
FEATURE: Configurable application token and application name

### PR DESCRIPTION
**Pull request by robertlemke**
This change introduces two news setting. One which allows for configuring
what is exposed via the default "X-Powered-By" HTTP header and a
second one which allows for configuring a human-readable application
name.

Previously, the header ("X-Flow-Powered") exposed the full version number
of Flow which may be undesirable in certain setups. Through the new
setting `core.applicationToken` now one of three options can be chosen:

- "Off" (no X-Powered-By header is sent)
- "ApplicationName" (the application name only, determined via the core.applicationKey setting)
- "MajorVersion" (the application name + major version, e.g. "Neos/2"
- "MinorVersion" (the application name + minor version, e.g. "Neos/2.1"

The new `core.applicationName` setting is used for defining a human-
friendly name, like "Neos" or "Flow", which is used for the `./flow` help
message and in the X-Powered-By header.

### CODE REVIEW

**Comment by robertlemke** [comment]
FYI, don't merge yet, tests need to be adjusted ...

File: TYPO3.Flow/Classes/TYPO3/Flow/Http/RequestHandler.php [hunk 1]
```diff
@@ -103,6 +104,7 @@ public function handleRequest()
 
         $this->boot();
         $this->resolveDependencies();
+        $this->addPoweredHeader($this->response);
```
Comment (kdambekalns) [inline TYPO3.Flow/Classes/TYPO3/Flow/Http/RequestHandler.php]: `addPoweredByHeader`?

**Comment by robertlemke** [comment]
:+1:

**Comment by kitsunet** [comment]
Looks good

**Comment by robertlemke** [comment]
So girls, how about this one? Or maybe a bit more pink?

**Comment by robertlemke** [comment]
Oh, and while you're at it - do you also like https://github.com/neos/neos-development-collection/pull/397 ?

**Comment by robertlemke** [comment]
But really, isn't this version much nicer than any of the earlier ones?

(switched back from X-Powered-By to X-Flow-Powered because in some exotic setups, people might rely on / be interested in the original X-Powere-By header sent by PHP)

**Comment by kdambekalns** [comment]
:+1:

**Comment by robertlemke** [comment]
:+1:

**Pull request merged**